easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -139,13 +139,20 @@ def build_type(self):
             build_type = 'Debug' if self.toolchain.options.get('debug', None) else 'Release'
         return build_type
 
-    def prepend_config_opts(self, config_opts):
-        """Prepends configure options (-Dkey=value) to configopts ignoring those already set"""
+    def combine_config_opts(self, config_opts_default):
```

Comment:
I don't think this is a good approach to fix #2881, since it *removes* a method (`prepend_config_opts`) from a commonly used generic easyblock like `CMakeMake`.
Although we don't have any easyblocks in the central repository that derive from `CMakeMake` and call `prepend_config_opts`, sites could have their own easyblocks that do, so we shouldn't be breaking the API like this.

As mentioned, not updating `configopts` in place also has other downsides, like easyblocks deriving from `CMakeMake` not being able to inspect the modified `configopts`.